client/src/assets/js/app/typing-lesson.js

```diff
@@ -113,6 +115,24 @@
       isRunning: function() {
         return (this.state === LevelStates.Running);
       },
+      sortIncorrectChars: function() {
+        let charsCount = this.result.missed_chars_count;
+
+        for(let key in charsCount){
+          let incorrectCharCount = charsCount[key];
+          let incorrectPercent = (incorrectCharCount / this.totalWrongCount);
+          this.incorrectChars.push({
+            letter: key,
+            count: incorrectCharCount,
+            percent: (incorrectPercent * 100),
+            opacity: incorrectPercent,
```

Comment:
This shouldn't be in the model/data layer.  It knows too much about what the UI does.  Use `char.percent` in the UI to get the `opacity`